src/orquestra/sdk/_base/cli/_dorq/_arg_resolvers.py

```diff
@@ -103,6 +104,108 @@ def resolve(
         return selected_id
 
 
+class TaskInvIDResolver:
+    """
+    Finds task invocation ID. Assumes workflow run ID and config were already
+    resolved.
+    """
+
+    def __init__(
+        self,
+        wf_run_repo=_repos.WorkflowRunRepo(),
+        fn_name_prompter=_prompts.Prompter(),
+        task_inv_prompter=_prompts.Prompter(),
+    ):
+        self._wf_run_repo = wf_run_repo
+        self._fn_name_prompter = fn_name_prompter
+        self._task_inv_prompter = task_inv_prompter
+
+    def resolve(
+        self,
+        task_inv_id: t.Optional[TaskInvocationId],
+        fn_name: t.Optional[str],
+        wf_run_id: WorkflowRunId,
+        config: ConfigName,
+    ) -> TaskInvocationId:
+        if task_inv_id is not None:
+            # User passed task inv ID directly.
+            return task_inv_id
+
+        if fn_name is not None:
+            # User passed fn name directly.
+            resolved_fn_name = fn_name
+        else:
+            fn_names = self._wf_run_repo.get_task_fn_names(wf_run_id, config)
+            if len(fn_names) > 1:
+                resolved_fn_name = self._fn_name_prompter.choice(
+                    fn_names, message="Task function name"
+                )
+            else:
+                resolved_fn_name = fn_names[0]
+
+        inv_ids = self._wf_run_repo.get_task_inv_ids(
+            wf_run_id=wf_run_id,
+            config_name=config,
+            task_fn_name=resolved_fn_name,
+        )
+        if len(inv_ids) > 1:
+            resolved_inv_id = self._task_inv_prompter.choice(
```

Comment:
Thought: (Non-blocking but something to be aware) - it came out on our latest OWG that we might want to be prompt-less for some automated-usage. Given this logic, when user provides fn_name there is no way to combine other things to avoid prompt. This makes sense given the nature of this parameter, but I think its worth saying it loud.

Reply:
I think we have all the bases covered:
- `dorq task logs --fn-name foo` -> we prompt for config, wf run ID and task inv ID.
- `dorq task logs --fn-name foo --task-inv-id inv1` -> we prompt for config, and wf run ID.
- `dorq task logs --fn-name foo --task-inv-id inv1` -> we prompt for config, and wf run ID.
- `dorq task logs --wf-run-id wf1 --fn-name foo --task-inv-id inv1` -> we prompt for config (if we can't infer it from wfs DB cache).
- `dorq task logs --config ray --wf-run-id wf1 --fn-name foo --task-inv-id inv1` -> we don't prompt for anything
- `dorq task logs --config ray --task-run-id foobar` -> we don't prompt for anything